shared/js/typing_status.js.flow

```diff
@@ -0,0 +1,19 @@
+/**
+ * @flow strict
+ */
+
+"use strict";
+
+type RecipientUserIds = number[];
+
+type Worker = {|
+  get_recipient: () => RecipientUserIds | void,
+  is_valid_conversation: (RecipientUserIds | void) => boolean,
+  get_current_time: () => number, // as ms since epoch
+  notify_server_start: RecipientUserIds => void,
+  notify_server_stop: RecipientUserIds => void
+|};
```

Comment:
This feels like it should be an `interface`, or at least not `{| exact |}`: I don't think there's a need to prevent a caller of `handle_text_input` from passing in an object with internal state and/or private auxiliary functions.

(It would also be nice if these had a line or two of documentation each, although this almost certainly isn't the right file for it.)

Reply:
Hmm yeah maybe. I agree that that'd be the more precisely accurate type for this signature.

I wrote an exact object type mainly because I've had enough experiences with Flow behaving in surprising ways on inexact object types :-/ that I generally avoid them unless there's a reason they really need to be there. At the existing call site in `typing.js`, the need for state and private auxiliary functions is handled in a classic JS idiom: the "worker" really is a plain object with just these properties, but the functions that are those properties' values close over a shared outer scope.


> (It would also be nice if these had a line or two of documentation each, although this almost certainly isn't the right file for it.)

Yeah, agreed on both counts. As you may have seen, the `typing_status.js` file has this:
```
/*

    Our parent should pass in a worker object with the following
    callbacks:

        notify_server_start
        notify_server_stop
        get_recipient
        get_current_time
        is_valid_conversation

    See typing.js for the implementations of the above. (Our
    node tests also act as workers and will stub those functions
    appropriately.)
*/
```

It would be good to have a line or two of documentation for each of these rather than "see implementation". Particularly for `is_valid_conversation`, which I've been unable to work out a particularly coherent semantics for at all.

---

We'll have plenty of opportunity to revise this interface -- in fact I have a draft series to refactor the `typing_status` API, which I expect to pick up once these infrastructural changes are in. So I hope we can get the infrastructural changes merged without blocking on the details of the specific module's API.